### PR TITLE
bumped min version of Twig to 1.28

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | see twigphp/Twig#2236
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by javiereguiluz** [comment]
What about using this great feature in Form, the component that needs it most? They even define [a method called renderBlock()](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/FormRenderer.php#L95)

File: src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php [hunk 1]
```diff
@@ -67,13 +67,15 @@ public function getName(Profile $profile, $panel)
      *
      * @param Profile $profile
      *
-     * @return array
+     * @return Twig_Template[]
+     *
+     * @deprecated not used anymore internally
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php]: shouldn't it throw a deprecation warning ?

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php [hunk 2]
```diff
@@ -200,7 +200,7 @@ public function toolbarAction(Request $request, $token)
         return new Response($this->twig->render('@WebProfiler/Profiler/toolbar.html.twig', array(
             'position' => $position,
             'profile' => $profile,
-            'templates' => $this->getTemplateManager()->getTemplates($profile),
+            'templates' => $this->getTemplateManager()->getNames($profile),
```
Comment (xabbuh) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: `getNames()` must become `public` then

File: src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php]: Agreed, but not in Symfony 2.7. This must be added after merging this to `master`.

Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Profiler/TemplateManager.php]: Agreed with @xabbuh, but this class should probably be marked as being internal as well in master.

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php (hunk 2 above)
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php]: sure :)

**Comment by fabpot** [comment]
@javiereguiluz Unfortunately the logic in the Form integration is complex and cannot use this technique as we are manually passing blocks to `renderBlock()`.

**Comment by fabpot** [comment]
Updated based on the latest changes for Twig 1.28.

**Comment by xabbuh** [comment]
:+1:

Status: Reviewed

**Pull request merged**